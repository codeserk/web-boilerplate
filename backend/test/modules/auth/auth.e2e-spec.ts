import { TestApplication } from '../../fixtures/application.fixture'

const loginMatcher = {
  // Access tokens are JWT
  token: expect.any(String),

  user: {
    id: expect.any(String),
  },
}

describe('/auth', () => {
  let app: TestApplication

  beforeAll(async () => {
    // Create the app
    app = await TestApplication.create()
  })

  afterAll(async () => {
    await TestApplication.tearDown(app)
  })

  beforeEach(() => {
    jest.clearAllMocks()
    TestApplication.clearDatabase(app)
  })

  afterEach(() => {
    app.logout()
  })

  describe('POST /login', () => {
    it('should login with a user', async () => {
      // Act
      // Creates a user and performs the login
      const response = await app.login()

      // Assert
      expect(response).toMatchSnapshot(loginMatcher)
    })
  })
})
