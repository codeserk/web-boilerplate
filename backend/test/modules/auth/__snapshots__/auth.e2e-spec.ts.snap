// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/auth POST /login should login with a user 1`] = `
Object {
  "token": Any<String>,
  "user": Object {
    "id": Any<String>,
    "role": "admin",
    "username": "test",
  },
}
`;
